nucleus/common/internal-api/src/main/java/org/glassfish/internal/deployment/JandexIndexer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) [2025] Payara Foundation and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://github.com/payara/Payara/blob/main/LICENSE.txt
+ * See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package org.glassfish.internal.deployment;
+
+import org.glassfish.api.deployment.DeploymentContext;
+import org.glassfish.api.deployment.archive.ReadableArchive;
+import org.jboss.jandex.Index;
+import org.jvnet.hk2.annotations.Contract;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+@Contract
+public interface JandexIndexer {
+    void index(DeploymentContext deploymentContext) throws IOException;
+    void reindex(DeploymentContext deploymentContext) throws IOException;
+    boolean isJakartaEEApplication(DeploymentContext deploymentContext) throws IOException;
+    Index getRootIndex(DeploymentContext deploymentContext);
```

Comment:
You might want to avoid spreading the JBoss classes like i.e. `Index` around in the Payara codebase.

Reply:
ha, that's kinda the point of this PR.
The issue is that current annotation scanning framework is not made for CDI, is slow, and can't be cached.
Jandex is a small, separate project, has no dependencies, and is tailor made for CDI.
While it did come from JBoss, it's no longer embedded in that ecosystem and can be used freely.

Any effort to isolate `Index` class will result in an unnecessary wrapper that would not add any value.

Reply:
It would result in a wrapper class. I agree. But I would call it a useful one. It allows to control and limit the interface towards payara code and allows to switch technology if needed/wanted.

Reply:
And you could have convenience methods useful for the payaras interaction with the index.

Reply:
Let's see where this goes... I think it's only used in one place anyway

Reply:
Had to create a wrapper class anyway, so this has been resolved.